### PR TITLE
Kyria: Updated personal keymap, added asapjockey's keymap

### PR DESCRIPTION
**Pull request by splitkb**
## Description

These change add a keymap for a customer, and make some changes to my own keymap. These are purely keymap changes and do not contain any other additions.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kyria/keymaps/asapjockey/keymap.c [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+#define MODS_SHIFT_MASK (MOD_BIT(KC_LSHIFT)|MOD_BIT(KC_RSHIFT))
+
+enum layers {
+    QWERTY = 0,
```
Comment (drashna) [inline keyboards/kyria/keymaps/asapjockey/keymap.c]: The 0 isn't necessary, as it should start at 0. 

```suggestion
    QWERTY,
```

File: keyboards/kyria/keymaps/asapjockey/keymap.c [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+#define MODS_SHIFT_MASK (MOD_BIT(KC_LSHIFT)|MOD_BIT(KC_RSHIFT))
```
Comment (drashna) [inline keyboards/kyria/keymaps/asapjockey/keymap.c]: This isn't needed.  We have `MOD_MASK_SHIFT` defined in the core: 
https://github.com/qmk/qmk_firmware/blob/1b7a003d842c76cd141af726c13e2af1a4458bac/tmk_core/common/keycode.h#L49-L63

```suggestion
```

File: keyboards/kyria/keymaps/asapjockey/keymap.c [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+//       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+//                                  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+//     ),
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case KC_LBR: {
+            if (record->event.pressed) {
+                if (keyboard_report->mods & MODS_SHIFT_MASK) {
```
Comment (drashna) [inline keyboards/kyria/keymaps/asapjockey/keymap.c]:
```suggestion
                if (get_mods() & MOD_MASK_SHIFT) {
```

Also, if you ONLY want shift, then: 
```suggestion
                if ((get_mods() & MODS_SHIFT_MASK) == MOD_MASK_SHIFT) {
```

File: keyboards/kyria/keymaps/asapjockey/keymap.c [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+                    register_code(KC_9);
+                } else {
+                    register_code(KC_LBRC);
+                }
+            } else { // Release the key
+                if (keyboard_report->mods & MODS_SHIFT_MASK) {
```
Comment (drashna) [inline keyboards/kyria/keymaps/asapjockey/keymap.c]:
```suggestion
                if ((get_mods() & MODS_SHIFT_MASK) == MOD_MASK_SHIFT) {
```

File: keyboards/kyria/keymaps/asapjockey/keymap.c [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+                } else {
+                    unregister_code(KC_LBRC);
+                }
+            }
+            return false;
+            break;
+        }
+        case KC_RBR: {
+            if (record->event.pressed) {
+                if (keyboard_report->mods & MODS_SHIFT_MASK) {
```
Comment (drashna) [inline keyboards/kyria/keymaps/asapjockey/keymap.c]:
```suggestion
                if ((get_mods() & MODS_SHIFT_MASK) == MOD_MASK_SHIFT) {
```

File: keyboards/kyria/keymaps/asapjockey/keymap.c [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+                    register_code(KC_0);
+                } else {
+                    register_code(KC_RBRC);
+                }
+            } else { // Release the key
+                if (keyboard_report->mods & MODS_SHIFT_MASK) {
```
Comment (drashna) [inline keyboards/kyria/keymaps/asapjockey/keymap.c]:
```suggestion
                if ((get_mods() & MODS_SHIFT_MASK) == MOD_MASK_SHIFT) {
```

Comment (Duckle29) [reply keyboards/kyria/keymaps/asapjockey/keymap.c]: It should be MODS_MASK_SHIFT in the bit-wise and right?

**Comment by splitkb** [comment]
I've received a report that this keymap isn't working yet, please await further changes by me.

**Comment by splitkb** [comment]
Should be good to go after the automated check by Travis.

File: keyboards/kyria/keymaps/asapjockey/keymap.c [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+    } else {
+        render_kyria_logo();
+    }
+}
+#endif
+
+#ifdef ENCODER_ENABLE
+void encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) {
+        switch (biton32(layer_state)) {
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/asapjockey/keymap.c]:
```suggestion
        switch (get_highest_layer(layer_state)) {
```

File: keyboards/kyria/keymaps/asapjockey/keymap.c [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+                 if (clockwise) { // Subscript: CTRL + =
+                    tap_code16(C(KC_EQL));
+                }
+                break;
+            default:
+                // Nothing
+                break;
+        }
+    } else if (index == 1) {
+        switch (biton32(layer_state)) {
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/asapjockey/keymap.c]:
```suggestion
        switch (get_highest_layer(layer_state)) {
```

File: keyboards/kyria/keymaps/asapjockey/keymap.c [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+            break;
+        case ADJUST:
+            oled_write_P(PSTR("Adjust\n"), false);
+            break;
+        default:
+            oled_write_P(PSTR("Undefined\n"), false);
+    }
+
+    // Host Keyboard LED Status
+    uint8_t led_usb_state = host_keyboard_leds();
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/asapjockey/keymap.c]:
```suggestion
    led_t led_state = host_keyboard_led_state();
```

File: keyboards/kyria/keymaps/asapjockey/keymap.c [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+    oled_write_P(IS_LED_ON(led_usb_state, USB_LED_NUM_LOCK)    ? PSTR("NUMLCK ") : PSTR("       "), false);
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/asapjockey/keymap.c]:
```suggestion
    oled_write_P(led_state.num_lock ? PSTR("NUMLCK ") : PSTR("       "), false);
```

File: keyboards/kyria/keymaps/asapjockey/keymap.c [hunk 11]
```diff
@@ -0,0 +1,312 @@
...
+    oled_write_P(IS_LED_ON(led_usb_state, USB_LED_CAPS_LOCK)   ? PSTR("CAPLCK ") : PSTR("       "), false);
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/asapjockey/keymap.c]:
```suggestion
    oled_write_P(led_state.caps_lock ? PSTR("CAPLCK ") : PSTR("       "), false);
```

File: keyboards/kyria/keymaps/asapjockey/keymap.c [hunk 12]
```diff
@@ -0,0 +1,312 @@
...
+    oled_write_P(IS_LED_ON(led_usb_state, USB_LED_SCROLL_LOCK) ? PSTR("SCRLCK ") : PSTR("       "), false);
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/asapjockey/keymap.c]:
```suggestion
    oled_write_P(led_state.scroll_lock ? PSTR("SCRLCK ") : PSTR("       "), false);
```

**Comment by fauxpark** [comment]
Just these small tweaks, otherwise LGTM

File: keyboards/kyria/keymaps/asapjockey/keymap.c [hunk 13]
```diff
@@ -0,0 +1,312 @@
...
+//       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+//                                  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+//     ),
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case KC_LBR: {
+            if (record->event.pressed) {
+                if ((get_mods() & MOD_MASK_SHIFT) == MOD_MASK_SHIFT) {
```
Comment (fauxpark) [inline keyboards/kyria/keymaps/asapjockey/keymap.c]: If I'm reading this right, `KC_9` will be registered only when *both* shifts are pressed, yes?

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**